### PR TITLE
Fixed stall telemetry issue.

### PR DESCRIPTION
**Pull request by byuarus**
Now we can dispose observables.
Cancel ServiceSubscribtionCalls on disposal.
Resubscribe if callResult is not .ok

### CODE REVIEW

File: Dronecode-SDK-Swift/Telemetry.swift [hunk 1]
```diff
@@ -381,289 +381,649 @@ public class Telemetry {
...
-                    let position = Position(latitudeDeg: response.position.latitudeDeg, longitudeDeg: response.position.longitudeDeg, absoluteAltitudeM: response.position.absoluteAltitudeM, relativeAltitudeM: response.position.relativeAltitudeM)
-
-                    observer.onNext(position)
+            var call: DronecodeSdk_Rpc_Telemetry_TelemetryServiceSubscribePositionCall?
+            
+            func subscribePosition() -> DronecodeSdk_Rpc_Telemetry_TelemetryServiceSubscribePositionCall? {
+                do {
+                    return try self.service.subscribePosition(positionRequest, completion: { (callResult) in
+                        if callResult.statusCode != .cancelled && callResult.statusCode != .ok {
+                            call = subscribePosition()
```
Comment (julianoes) [inline Dronecode-SDK-Swift/Telemetry.swift]: Do I understand it correctly that this is recursion and will grow the stack every time we have to re-subscribe?

**Comment by JonasVautherin** [comment]
Thanks a lot @byuarus, now I believe I understand a bit better how completions work :-).

I tried to benefit from RxSwift's `retry` operator to remove the recursion in your solution (as @julianoes pointed out, the recursion could be an issue if it tries to connect and fails forever, right?).

Please let me know what you think about it, I made a PR to your branch directly: https://github.com/Dronecode/DronecodeSDK-Swift/pull/90

File: Dronecode-SDK-Swift/Telemetry.swift (hunk 1 above)
Comment (byuarus) [reply Dronecode-SDK-Swift/Telemetry.swift]: It looks a little confusing but it's not a classic recursion, it's an asynchronous recursion as we assign a new function inside of the completion handler. It won't increase the stack of calls.

**Comment by JonasVautherin** [comment]
LGTM, I'll run some tests in the afternoon to confirm that it works on my side before I approve!

**Pull request merged**